### ISSUE TITLE
Error "AttributeError: 'str' object has no attribute '__name__'"

### ISSUE THREAD

**Issue by soimkim** [nexB/scancode-toolkit#2521]
### Description
When executing scancode, the following error message occurs.
```
$ scancode -h
Traceback (most recent call last):
  File "/home/worker/venv3/bin/scancode", line 5, in <module>
    from scancode.cli import scancode
  File "/home/worker/venv3/lib/python3.6/site-packages/scancode/cli.py", line 108, in <module>
    plugin_classes, plugin_options = PluginManager.load_plugins()
  File "/home/worker/venv3/lib/python3.6/site-packages/plugincode/__init__.py", line 207, in load_plugins
    mgr_setup = manager.setup()
  File "/home/worker/venv3/lib/python3.6/site-packages/plugincode/__init__.py", line 231, in setup
    self.manager.load_setuptools_entrypoints(entrypoint)
  File "/home/worker/venv3/lib/python3.6/site-packages/pluggy/manager.py", line 299, in load_setuptools_entrypoints
    plugin = ep.load()
  File "/home/worker/venv3/lib/python3.6/site-packages/importlib_metadata/__init__.py", line 166, in load
    module = import_module(match.group('module'))
  File "/usr/lib/python3.6/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "/home/worker/venv3/lib/python3.6/site-packages/summarycode/classify.py", line 58, in <module>
    class FileClassifier(PreScanPlugin):
  File "/home/worker/venv3/lib/python3.6/site-packages/summarycode/classify.py", line 108, in FileClassifier
    sort_order=50,
  File "/home/worker/venv3/lib/python3.6/site-packages/commoncode/cliutils.py", line 386, in __init__
    **kwargs
  File "/home/worker/venv3/lib/python3.6/site-packages/click/core.py", line 2467, in __init__
    super().__init__(param_decls, type=type, multiple=multiple, **attrs)
  File "/home/worker/venv3/lib/python3.6/site-packages/click/core.py", line 2039, in __init__
    self.type = convert_type(type, default)
  File "/home/worker/venv3/lib/python3.6/site-packages/click/types.py", line 1022, in convert_type
    return FuncParamType(ty)
  File "/home/worker/venv3/lib/python3.6/site-packages/click/types.py", line 158, in __init__
    self.name = func.__name__
AttributeError: 'str' object has no attribute '__name__'
````

### How To Reproduce

````
pip3 install scancode-toolkit
scancode -h
````

### System configuration

> For bug reports, it really helps us to know:

* What OS are you running on? Ubuntu 18.04
* What version of scancode-toolkit was used to generate the scan file?21.3.31
* What installation method was used to install/run scancode?pip

**Comment by jmbowman** [comment]
This looks like an incompatibility with click 8.0.0, which was just released 2 days ago.  Downgrading to click 7.1.2 seems to work until the code gets updated accordingly.

**Comment by ethan-ic** [comment]
Downgrading to click 7.1.2 and it works.
```
$ scancode -h
Traceback (most recent call last):
  File "/home/sit/venv/bin/scancode", line 5, in <module>
    from scancode.cli import scancode
  File "/home/sit/venv/lib/python3.8/site-packages/scancode/cli.py", line 108, in <module>
    plugin_classes, plugin_options = PluginManager.load_plugins()
  File "/home/sit/venv/lib/python3.8/site-packages/plugincode/__init__.py", line 207, in load_plugins
    mgr_setup = manager.setup()
  File "/home/sit/venv/lib/python3.8/site-packages/plugincode/__init__.py", line 231, in setup
    self.manager.load_setuptools_entrypoints(entrypoint)
  File "/home/sit/venv/lib/python3.8/site-packages/pluggy/manager.py", line 299, in load_setuptools_entrypoints
    plugin = ep.load()
  File "/usr/lib/python3.8/importlib/metadata.py", line 77, in load
    module = import_module(match.group('module'))
  File "/usr/lib/python3.8/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "/home/sit/venv/lib/python3.8/site-packages/summarycode/classify.py", line 58, in <module>
    class FileClassifier(PreScanPlugin):
  File "/home/sit/venv/lib/python3.8/site-packages/summarycode/classify.py", line 103, in FileClassifier
    PluggableCommandLineOption(('--classify',),
  File "/home/sit/venv/lib/python3.8/site-packages/commoncode/cliutils.py", line 385, in __init__
    super(PluggableCommandLineOption, self).__init__(
  File "/home/sit/venv/lib/python3.8/site-packages/click/core.py", line 2467, in __init__
    super().__init__(param_decls, type=type, multiple=multiple, **attrs)
  File "/home/sit/venv/lib/python3.8/site-packages/click/core.py", line 2039, in __init__
    self.type = convert_type(type, default)
  File "/home/sit/venv/lib/python3.8/site-packages/click/types.py", line 1022, in convert_type
    return FuncParamType(ty)
  File "/home/sit/venv/lib/python3.8/site-packages/click/types.py", line 158, in __init__
    self.name = func.__name__
AttributeError: 'str' object has no attribute '__name__'

$ pip install click==7.1.2
Collecting click==7.1.2
  Using cached click-7.1.2-py2.py3-none-any.whl (82 kB)
Installing collected packages: click
  Attempting uninstall: click
    Found existing installation: click 8.0.0
    Uninstalling click-8.0.0:
      Successfully uninstalled click-8.0.0
Successfully installed click-7.1.2

$ scancode -h
/home/sit/venv/lib/python3.8/site-packages/cluecode/copyrights.py:3122: FutureWarning: Possible set difference at position 3
  remove_tags = re.compile(
/home/sit/venv/lib/python3.8/site-packages/nltk/tag/sequential.py:540: FutureWarning: Possible nested set at position 7
  self._regexps = [(re.compile(regexp), tag,) for regexp, tag in regexps]
Usage: scancode [OPTIONS] <OUTPUT FORMAT OPTION(s)> <input>...

  scan the <input> file or directory for license, origin and packages and
  save results to FILE(s) using one or more output format option.

  Error and progress are printed to stderr.

Options:

  primary scans:
    -l, --license    Scan <input> for licenses.
    -p, --package    Scan <input> for package manifests and build scripts.
    -c, --copyright  Scan <input> for copyrights.

  other scans:
    -i, --info   Scan <input> for file information (size, checksums, etc).
    --generated  Classify automatically generated code files with a flag.
    -e, --email  Scan <input> for emails.
    -u, --url    Scan <input> for urls.

  scan options:
```

**Comment by pombredanne** [comment]
Thanks! I guess no good deed goes unpunished then, as I had removed an upper bound on click version to accommodate for possible conflicts that required  a v7 in https://github.com/nexB/scancode-toolkit/commit/2b986e7bdea0214bd832136d710709df9ef7e65e
Then 7.0.0 failed was a problem and needed skipping .... so I guess we can set an upper bound of `<8.0` for now?

**Comment by danielthieleog** [comment]
`pip install "click<8.0` installed `7.1.2`. That worked for me with python 3.9

**Comment by soimkim** [comment]
As the comment, I installed with click 7.1.2, so no error occurred.
Thanks for answering.

**Comment by pombredanne** [comment]
@soimkim actually this is a bug alright. See #2529  and https://github.com/nexB/commoncode/pull/19

**Comment by pombredanne** [comment]
This is fixed in v21.6.7 and up. Thanks!
